tensorflow/c/experimental/gradients/gradient_checker.cc

```diff
@@ -144,24 +135,42 @@ Status CalcNumericalGrad(AbstractContext* ctx, Model forward,
   // Numerical Grad Check
   for (int i = 0; i < num_elems; i++) {
     // Get relative epsilon value
-    float epsilon =
-        std::abs(theta_data[i] * 1e-4 + 1e-4);  // add 1e-4 to prevent div by 0
-    AbstractTensorHandlePtr two_eps =
-        GetScalarTensorHandleUtil(ctx, 2 * epsilon);
+    float epsilon = theta_data[i] == 0 ? 1e-4 : std::abs(theta_data[i] * 1e-4);
```

Comment:
when `theta_data[i] == -1`, `epsilon` will be 0 and it causes a lot of confusions. It took me a day to figure it out. In addition, I changed `ops::DivNoNan` to `ops::Div` to avoid dividing by zero.